04-rag-demo/README.md

```diff
@@ -0,0 +1,135 @@
+# RisingWave RAG Demo
+
+This is a demo of using RisingWave to build a RAG system to answer questions about RisingWave. The data of RAG is from the [RisingWave docs](https://docs.risingwave.com/). It stores the document contents as well as the embeddings of the documents. 
+
+When a user asks a question, the system will use the question embedding to find the most similar documents from the vector database, and then call LLM model and use the document contents to answer the question.
+
+## Setup
+
+You can use this 1-line command to start the RisingWave server.
+
+```bash
+curl -L https://risingwave.com/sh | sh
+```
+
+After the server started, create the necessary UDFs and tables.
+
+The following SQL creates a UDF `text_embedding` that uses OpenAI's API to embed a text into a vector. The function relies on the `fetch` API, which is introduced in the latest version of RisingWave.
+
+```sql
+create function text_embedding(t varchar) returns real[] language javascript as $$
+    export async function text_embedding(t) {
+        for (let retries = 0; retries < 3; retries++) {
+            try {
+                const response = await fetch("https://api.openai.com/v1/embeddings", {
+                    method: 'POST',
+                    headers: {
+                        'Authorization': `Bearer <your-openai-api-key>`,
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        input: t,
+                        model: "text-embedding-3-small"
+                    })
+                });
+                if (!response.ok) {
+                    const error = await response.json();
+                    throw new Error(`OpenAI API error: ${error.error?.message || 'Unknown error'}`);
+                }
+                const data = await response.json();
+                return data.data[0].embedding;
+            } catch (error) {
+            }
+        }
+        return null;
+    }
+$$ WITH (
+    async = true,
+    batch = false
```

Comment:
I think we can use `batch = true` now?

Reply:
Haha, I found it hits https://github.com/risingwavelabs/risingwave/issues/20473